openedx_proversity_reports/api/v0/views.py

```diff
@@ -73,159 +71,13 @@ def post(self, request):
-            json_response["message"] = message
-            logger.info(message)
-            return JsonResponse(
-                json_response,
-                status=status.HTTP_400_BAD_REQUEST
-            )
-
-        block_report_filter = request.data.get("block_report_filter", ["vertical"])
-
-        task = generate_completion_report.delay(courses, block_report_filter)
-        state_url = request.build_absolute_uri(reverse('proversity-reports:api:v0:completion-report-data'))
-
-        logger.info("The task with id = %s has been initialize.", task.id)
-
-        json_response["success"] = True
-        json_response["state_url"] = "{}?task_id={}".format(state_url, task.id)
-        json_response["message"] = "The task with id = {} has been initialize.".format(task.id)
-
-        return JsonResponse(json_response, status=status.HTTP_202_ACCEPTED)
-
-
-class CompletionReportView(APIView):
-    """
-    This class verifies the status for the given task id and returns the result.
-    """
-
-    authentication_classes = (
-        OAuth2Authentication,
-        get_jwt_authentication(),
-    )
-    permission_classes = (permissions.IsAuthenticated, get_staff_or_owner())
-
-    def get(self, request):
-        """
-        This method starts a celery task that generates a report with the information about
-        the required activities and its state.
-
-        **Params**
-
-            task_id: the identifier for the task
-
-
-        **Example Requests**:
-
-            GET /proversity-reports/api/v0/completion-report-data?task_id=4309f98a-b7e9-48e6-b9a7-996e640ece2e/
-
-        **Response Values**:
-
-            status: task status.
-            result: the task result.
-
-        **Example Response**:
-            {
-               "status":"SUCCESS",
-               "result":{
-                  "course-v1:edx+cs101+2019":[
-                     {
-                        "username":"edx",
-                        "user_id":2,
-                        "vertical":{
-                           "Problems-Unit-2":false,
-                           "Rocketchat Teams-Unit":false,
-                           "Problem-Unit":true,
-                           "Subsection-Unit":true,
-                           "Html-Unit":true,
-                           "Problems-Unit-1":false,
-                           "Html-Unit-2":true,
-                           "Problems-Unit":false,
-                           "Discussion-Unit":true,
-                           "Html-Unit-3":true,
-                           "Html-Unit-1":true,
-                           "Video-Unit":false
-                        }
-                     }
-                  ]
-               }
-            }
-        """
-        task_id = request.GET.get("task_id")
+        report_name = report_name.replace('-', '_')
+        task_module = import_module(SUPPORTED_TASKS_MODULE)
+        task = getattr(task_module, report_name, None)
```

Comment:
Getting the task module could be a unique function, maybe in the future we need to add more logic to this. @andrey-canon

Reply:
done